modules/tinymce/src/plugins/autolink/main/ts/core/Keys.ts

```diff
@@ -203,21 +200,17 @@ const handleSpacebar = (editor: Editor): void => {
 
 const handleBracket = handleSpacebar;
 
-const handleEnter = (editor: Editor, e: EditorEvent<KeyboardEvent>): void => {
+const handleEnter = (editor: Editor): void => {
   const result = parseCurrentLine(editor, -1);
   if (Type.isNonNullable(result)) {
```

Comment:
Can you briefly explain why we had the mceInsertNewline before, and why we are removing it?

Reply:
Because you need to ensure the newline is inserted before doing the replacement if you want the undo to only undo the link change and not the newline instead.

Reply:
To word that differently, it was only added originally to fix the undo stack, but that's part of what caused the issues here so for now we're just abandoning it and undo will first undo the new line then the url conversion.

Reply:
K. That makes sense.